### ISSUE TITLE
How does it work?

### ISSUE THREAD

**Issue by vjpr** [jhmaster2000/node-bun#4]
I couldn't figure it out exactly. Do you use `--experimental-loader`?

EDIT: Found the entry point here: https://github.com/jhmaster2000/node-bun/blob/master/src/cli.ts

**Comment by jhmaster2000** [comment]
The CLI is not really an entry point, it is just a wrapper to simplify the process of passing required flags such as `--loader` and `--expose-gc` (among others) which are required to achieve runtime features of bun (and a select few dependent polyfills), it is by no means necessary to use the CLI and passing the same flags directly to `node` binary would have the same effect.

Going a step further, for the majority of the actual polyfills to be used as a library standpoint, no flags are actually needed, and most will work by importing the actual "library" entry point file: https://github.com/jhmaster2000/node-bun/blob/master/src/polyfills.ts, of course without the runtime features, which includes the TS runtime, you will need to import a transpiled version and not directly the .ts source file, unless using some other node TS runtime like `ts-node`.

So to sum up: yes, `--experimental-loader` is used, in order to implement the runtime-level features such as TS runtime, CJS resolution for imports (formerly `--experimental-specifier-resolution=node`, which got removed in favor of forcing people to do it themselves with loaders), support for `bun:*` modules imports, Bun-exclusive `import.meta` properties, among others. However non-runtime-level features such as most of the actual functions polyfilled don't require the loader or any runtime flags. A notable exception to the polyfills is `Bun.gc()` which depends on `--expose-gc` as otherwise node exposes to mechanism to control the garbage collector and thus the polyfill is impossible.

**Comment by vjpr** [comment]
Thanks! This is great project and very necessary for the Bun ecosystem.

I maintain my own typescript loader and the library-mode will be very helpful.
